### DISCUSSION TITLE
Compilations without Albumartist are listed as separate Albums in Browse View

### DISCUSSION THREAD

**Discussion by smeikx** [jcorporation/myMPD#742] [Q&A] [answered]
myMPD version: 9.2.4

In the _Browse_ view, viewing albums, sorted by album, if an album consists of tracks by different artists, and each track has only the _artist_ set, but not the _albumartist,_ each track is listed as its own album. I expected to see only one entry.

![overview](https://user-images.githubusercontent.com/6908653/165785110-f3925686-f823-4e82-a971-c33a933e36f6.png)

If I select one of the albums/tracks and click on _Album details,_ I see this:

![album-details](https://user-images.githubusercontent.com/6908653/165785155-25436b78-dea7-4ef1-8692-b01baa14ab51.png)

Notice that the artist is listed as _albumartist_. This also happens when I click on _Song details:_

![song-details](https://user-images.githubusercontent.com/6908653/165785235-73843796-74ae-4814-9590-d7dd6ffd7f88.png)

I do not know if this is relevant to the actual issue or if it is an issue on its own.

Also, here is a screenshot of the song’s tags in [EasyTAG](https://wiki.gnome.org/Apps/EasyTAG):

![tags](https://user-images.githubusercontent.com/6908653/165785310-e47b48b0-7148-4902-a433-bbf0aeab6e3f.png)

**Server platform (please complete the following information):**
 - OS: Alpine Linux
 - OS version 3.15
 - Platform: aarch64
 - MPD version: 0.23.4

**Client platform (please complete the following information):**
I tested it with Firefox on Linux and Safari on iOS, both with the latest version, but I doubt the problem is in the front end.

Please let me know if the provided information is insufficient and if I can help in any way.

Besten Dank!

**Comment by jcorporation** [comment, marked as answer]
You should set the AlbumArtist tag for compilations. Grouping songs to albums is done by combining albumartist + album. If albumartist is not present myMPD falls back to artist.

**Reply by smeikx** [reply]
Hm, I would have to edit quite a few files. Also, I would have to resort to something like _Various Artists_ or _Compilation_ which irks me because these are not actually artist names.

But considering that it is not possible to group tracks as an album using the _Artist_ and _Album_ tags alone (there are multiple albums with the same name), I understand why _Album Artist_ is necessary.

Also, many players assume the presence of _Album Artist_ (even though ID3v2 doesn’t label this tag this way), so I might as well just give in.

Still, I don’t think that _Album Artist_ should show the _Artist_ in the album/song details if the tag is actually not set.

**Reply by jcorporation** [reply]
Right, the silent fallback from AlbumArtist to Artist is a bit unexpected, but there are some dependencies for this behavior. Best way is to maintain the AlbumArtist tag or to disable the tag in MPD.

**Reply by jcorporation** [reply]
> Still, I don’t think that Album Artist should show the Artist in the album/song details if the tag is actually not set.

This will be fixed with v9.4.0 release.
